pkg/ext-proc/backend/provider.go

```diff
@@ -217,3 +227,40 @@ func (p *Provider) refreshMetricsOnce() error {
 	}
 	return errs
 }
+
+func (p *Provider) flushPrometheusMetricsOnce() {
+	klog.V(logutil.DEFAULT).Infof("Flushing Prometheus Metrics")
+
+	pool, _ := p.datastore.getInferencePool()
+	if pool == nil {
+		// No inference pool or not initialize.
+		return
+	}
+
+	var kvCacheTotal float64
+	var queueTotal int
+
+	podTotalCount := 0
+	pods, err := p.datastore.getPods()
+	if err != nil {
+		klog.V(logutil.DEFAULT).Infof("Couldn't list pods: %v", err)
+		return
+	}
+
+	for _, pod := range pods {
+		if !podIsReady(pod) {
```

Comment:
we can assume that the pods in podMetrics are recent, so no need to check for readiness and we can iterate over the map directly

Reply:
Done.